crates/biome_js_analyze/src/assists/source/sort_jsx_props.rs

```diff
@@ -0,0 +1,132 @@
+use std::{borrow::Cow, cmp::Ordering, iter::zip};
+
+use biome_analyze::{
+    context::RuleContext, declare_source_rule, ActionCategory, Ast, Rule, RuleAction, RuleSource,
+    RuleSourceKind, SourceActionKind,
+};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_syntax::{AnyJsxAttribute, JsxAttribute, JsxAttributeList};
+use biome_rowan::{AstNode, BatchMutationExt};
+
+use crate::JsRuleAction;
+
+declare_source_rule! {
+    /// Enforce props sorting in JSX elements.
+    ///
+    /// This rule checks if the JSX props are sorted in a consistent way.
+    /// Props are sorted alphabetically.
+    /// A spread prop resets the sorting order.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// <Hello lastName="Smith" firstName="John" />;
+    /// ```
```

Comment:
It definitely makes sense to add an example with spread props, so users can understand how it works. Maybe two spread props should be enough